src/coreclr/jit/lowerxarch.cpp

```diff
@@ -9170,19 +9176,30 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
                         // Swap the operands here to make the containment checks in codegen significantly simpler
                         std::swap(node->Op(1), node->Op(2));
                     }
-                    else if (supportsOp2RegOptional)
+                    else if (supportsOp1RegOptional)
                     {
-                        MakeSrcRegOptional(node, op2);
+                        if (supportsOp2RegOptional)
+                        {
+                            GenTree* regOptionalOperand = PreferredRegOptionalOperand(op1, op2);
+                            MakeSrcRegOptional(node, regOptionalOperand);
+
+                            if (regOptionalOperand == op1)
+                            {
+                                // Swap the operands here to make the containment checks in codegen simpler
+                                std::swap(node->Op(1), node->Op(2));
```

Comment:
What is the idea behind the swap? We want the `regOptional` node as the last child?

Reply:
By swapping the operands, we greatly simplify the containment checks in codegen as we only have to have a path checking if `op2` is contained. Otherwise, we have to have logic to check if either `op1` or `op2` is contained and handle that difference.

Functionally the contained op is emitted as the final operand as well, so this makes it more consistent with the overall expected output for hardware intrinsics